category/fold.lean

```diff
@@ -0,0 +1,93 @@
+
+import data.list.basic
+import category.traversable.instances
+
+universes u v
+
+def foldl (α : Type u) (β : Type v) := α → α
+def foldr (α : Type u) (β : Type v) := α → α
```

Comment:
Hm, not sure I like this being in the root namespace. How about something slightly more unique like `tfoldl`?

Reply:
I could put them in `functor` as well, similarly to `comp`